lib/ansible/modules/network/mlnxos/mlnxos_interface.py

```diff
@@ -0,0 +1,402 @@
+short_description: Manage Interfaces on Mellanox MLNX-OS network devices
+description:
+  - This module provides declarative management of Interfaces
+    on Mellanox MLNX-OS network devices.
+notes:
+options:
+  name:
+    description:
+      - Name of the Interface.
+    required: true
+  description:
+    description:
+      - Description of Interface.
+  enabled:
+    description:
+      - Interface link status.
+  speed:
+    description:
+      - Interface link speed.
+  mtu:
+    description:
+      - Maximum size of transmit packet.
+  aggregate:
+    description: List of Interfaces definitions.
+  duplex:
+    description:
+      - Interface link status
+    default: auto
+    choices: ['full', 'half', 'auto']
+  tx_rate:
+    description:
+      - Transmit rate in bits per second (bps).
+  rx_rate:
+    description:
+      - Receiver rate in bits per second (bps).
+  delay:
+    description:
+      - Time in seconds to wait before checking for the operational state on
+        remote device. This wait is applicable for operational state argument
+        which are I(state) with values C(up)/C(down).
+    default: 10
+  purge:
+    description:
+      - Purge Interfaces not defined in the aggregate parameter.
+        This applies only for logical interface.
+    default: no
+  state:
+    description:
+      - State of the Interface configuration, C(up) means present and
+        operationally up and C(down) means present and operationally C(down)
+    default: present
+    choices: ['present', 'absent', 'up', 'down']
+"""
+
+EXAMPLES = """
+- name: configure interface
+  mlnxos_interface:
+      name: Eth1/2
+      description: test-interface
+      speed: 100 GB
+      mtu: 512
+
+- name: make interface up
+  mlnxos_interface:
+    name: Eth1/2
+    enabled: True
+
+- name: make interface down
+  mlnxos_interface:
+    name: Eth1/2
+    enabled: False
+
+- name: Check intent arguments
+  mlnxos_interface:
+    name: Eth1/2
+    state: up
+
+- name: Config + intent
+  mlnxos_interface:
+    name: Eth1/2
+    enabled: False
+    state: down
+"""
+
+RETURN = """
+commands:
+  description: The list of configuration mode commands to send to the device.
+  returned: always
+  type: list
+  sample:
+    - interface Eth1/2
+    - description test-interface
+    - mtu 512
+"""
+
+from copy import deepcopy
+import re
+from time import sleep
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.network.common.utils import conditional, \
+    remove_default_spec
+
+from ansible.module_utils.network.mlnxos.mlnxos import BaseMlnxosModule, \
+    get_interfaces_config
+
+
+class MlnxosInterfaceModule(BaseMlnxosModule):
+    ETH_IF_NAME_REGEX = re.compile(r'^Eth(\d\/\d+(|\/\d))$')
+
+    @classmethod
+    def _get_element_spec(cls):
+        return dict(
+            name=dict(type='str'),
+            description=dict(),
+            speed=dict(choices=['1G', '10G', '25G', '40G', '50G', '56G', '100G']),
```

Comment:
Needs adding to `DOCUMENTATION` under `choices:` see http://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#documentation-block